core/protocols/aws-query-protocol/src/main/java/software/amazon/awssdk/protocols/query/internal/marshall/ListQueryMarshaller.java

```diff
@@ -26,18 +26,29 @@
  */
 @SdkInternalApi
 public class ListQueryMarshaller implements QueryMarshaller<List<?>> {
+    private static final PathResolver AWS_QUERY_PATH_RESOLVER = (path, i, listTrait) ->
+        listTrait.isFlattened() ?
+        String.format("%s.%d", path, i + 1) :
+        String.format("%s.%s.%d", path, listTrait.memberFieldInfo().locationName(), i + 1);
```

Comment:
Maybe unpopular opinion: I preferred this in-line, because I like to read the logic in context with the method. Here I can't tell what either is doing without scrolling up and down.